x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/agent_policy_advanced_fields/index.tsx

```diff
@@ -317,13 +319,31 @@ export const AgentPolicyAdvancedOptionsContent: React.FunctionComponent<Props> =
           }
         >
           <EuiSwitch
-            label={i18n.translate('xpack.fleet.agentPolicyForm.tamperingSwitchLabel', {
-              defaultMessage: 'Prevent agent tampering',
-            })}
-            checked={agentPolicy.is_protected ?? false}
+            label={
+              <>
+                <FormattedMessage
+                  id="xpack.fleet.agentPolicyForm.tamperingSwitchLabel"
+                  defaultMessage="Prevent agent tampering"
+                />{' '}
+                {!policyHasEndpointSecurity(agentPolicy) && (
```

Comment:
Can we store and memoize the `!policyHasEndpointSecurity(agentPolicy)` in a const and use it below?

Reply:
Same thing for the `isAgentPolicy...`